core/src/main/java/org/apache/iceberg/rest/http/RequestResponseSerializers.java

```diff
@@ -0,0 +1,162 @@
+    @Override
+    public void serialize(Namespace namespace, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+      gen.writeString(namespace.toString());
+    }
+  }
+
+  public static class TableIdentifierDeserializer extends JsonDeserializer<TableIdentifier> {
+    @Override
+    public TableIdentifier deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return TableIdentifier.parse(jsonNode.asText());
+    }
+  }
+
+  public static class TableIdentifierSerializer extends JsonSerializer<TableIdentifier> {
+    @Override
+    public void serialize(TableIdentifier identifier, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeString(identifier.toString());
+    }
+  }
+
+  public static class SchemaDeserializer extends JsonDeserializer<Schema> {
+    @Override
+    public Schema deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      Schema schema = SchemaParser.fromJson(jsonNode);
+      context.setAttribute("schema", schema);
+      return schema;
+    }
+  }
+
+  public static class SchemaSerializer extends JsonSerializer<Schema> {
+    @Override
+    public void serialize(Schema schema, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      SchemaParser.toJson(schema, gen);
+    }
+  }
+
+  public static class PartitionSpecSerializer extends JsonSerializer<PartitionSpec> {
+    @Override
+    public void serialize(PartitionSpec partitionSpec, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      PartitionSpecParser.toJson(partitionSpec, gen);
+    }
+  }
+
+  public static class PartitionSpecDeserializer extends JsonDeserializer<PartitionSpec> {
+    @Override
+    public PartitionSpec deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+
+      Schema schema = (Schema) context.getAttribute("schema");
+
+      return PartitionSpecParser.fromJson(schema, jsonNode);
+    }
+  }
+
+  public static class SortOrderSerializer extends JsonSerializer<SortOrder> {
+    @Override
+    public void serialize(SortOrder sortOrder, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      SortOrderParser.toJson(sortOrder, gen);
+    }
+  }
+
+  public static class SortOrderDeserializer extends JsonDeserializer<SortOrder> {
+    @Override
+    public SortOrder deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      Schema schema = (Schema) context.getAttribute("schema");
+
+      return SortOrderParser.fromJson(schema, jsonNode);
+    }
+  }
+
+  public static class TableMetadataSerializer extends JsonSerializer<TableMetadata> {
```

Comment:
We'll need a deserializer as well. Currently that depends on having a metadata file. I have a few ideas around this and will push something during my next round of clean up on this. Possibly placing it in a separate PR as well.

Eventually, I plan on breaking this into smaller PRs to be more easily digested, but this is to provide a larger picture overview.

Reply:
Yeah, there's no need to only read from a metadata file.